app/models/grda_warehouse/source_client_view_accessor.rb

```diff
@@ -0,0 +1,53 @@
+# Provides view-optimized access to source client data for destination clients.
+# Handles both efficient data loading and formatted output for views, reducing
+# N+1 queries while providing convenient access to client information.
+#
+class GrdaWarehouse::SourceClientViewAccessor
+  # override inspect to prevent clutter exceptions
+  def inspect
+    object_id
+  end
+
+  # @param user [User] The current user accessing the client data
+  def initialize(user:)
+    @user = user
+    @source_clients = {}
+  end
+
+  # Retrieves source clients for a given destination client
+  # @param client [DestinationClient] The destination client to look up
+  # @return [Array<HudClient>] Array of source client records associated with the destination client
+  def source_clients(client)
+    key = client.id
+    preload_source_clients([client]) unless @source_clients.key?(key)
+    @source_clients[key] || []
+  end
+
+  # @param client [DestinationClient] The destination client to get names for
+  def client_names(client)
+    GrdaWarehouse::SourceClientNameSet.new(
+      destination_client: client,
+      source_clients: source_clients(client),
+      user: @user,
+    )
+  end
+
+  def preload_source_clients(clients)
```

Comment:
Should this method name indicate it is only returning those that are "searchable_to" the user?  Sometimes people can search for a client that they don't have access to through `clients_source_visible_to`.  

Since this returns the full client objects, it feels a bit dangerous.  Maybe it should be:

`preload_source_clients(clients, context: :searchable)` and `preload_source_clients(clients, context: :viewable)` to indicate what context the returned data should be used in.

Reply:
oh I see where you got the idea the return type was meaningful, I had some old incorrect code in the client_table partial. I'll fix that up

Reply:
The bit on line 42-43 below:

```
clients = GrdaWarehouse::Hud::Client.arbiter(@user).
clients_source_searchable_to(@user, client_ids: source_client_ids)
```

 is what makes me think we should rename.  

I like your options above. `searchable_client_names` and `searchable_clients` but I think `preload_source_clients` should be `preload_searchable_source_clients` because of the use of `arbiter.clients_source_searchable_to`.

Reply:
Alternatively, if `preload_source_clients` were private, that would suffice.